easybuild/easyblocks/b/bazel.py

```diff
@@ -119,7 +119,18 @@ def prepare_step(self, *args, **kwargs):
         """Setup bazel output root"""
         super(EB_Bazel, self).prepare_step(*args, **kwargs)
         self.bazel_tmp_dir = tempfile.mkdtemp(suffix='-bazel-tmp', dir=self.builddir)
-        self.output_user_root = tempfile.mkdtemp(suffix='-bazel-root', dir=self.builddir)
+        self._make_output_user_root()
+
+    def _make_output_user_root(self):
+        self._output_user_root = tempfile.mkdtemp(suffix='-bazel-root', dir=self.builddir)
```

Comment:
```suggestion
        self._output_user_root = tempfile.mkdtemp(suffix='-bazel-root')
```
I tried just removing the `dir` option to `mkdtemp` and that works just fine, it places the dir in the EB temp dir and the sanity check succeeds.

Reply:
```
== sanity checking...
  >> file 'bin/bazel' found: OK
  >> loading modules: Bazel/3.7.2-GCCcore-11.2.0...
  >> running command 'bazel --output_user_root=/tmp/eb-u1musahf/tmpzq815rsj-bazel-root --help' ...
  >> result for command 'bazel --output_user_root=/tmp/eb-u1musahf/tmpzq815rsj-bazel-root --help': OK
```

Reply:
I would vote for adding the first option: check if it exists, if yes use it, if not just use the temp dir.

Creating the builddir could easily trigger avoidable issues (like having to set the buildpath), and there's no real benefit.

Reply:
Done: 139c56e0d4939573836ba7a535dab1e32e15bba1